### DISCUSSION TITLE
Pac file

### DISCUSSION THREAD

**Discussion by dev-serena** [justcoding121/Titanium-Web-Proxy#842] [General]
I like ti migrate from fiddler core to titanium but my app monitor only one domain name.

We can use :
if (!hostName.Contains("mydomain.com"))
     {
        e.DecryptSsl = false;
     }
But it only not decode ssl all tracking still tunnel to titanium , memory size increase and maybe connection speed decrease. I like to use titanium only with this domain.

With fiddler i use .HookUsingPACFile() but with titanium is it a way to use https://127.0.0.1/config.pac?

**Comment by justcoding121** [comment]
`DecryptSsl = false` still tunnels through Titanium ? it just doesn't decrypt. To *bypass* the proxy for other hosts (like Fiddler's PAC), put a PAC on the client.

You can serve one from Titanium:

```csharp
proxyServer.BeforeRequest += async (sender, e) =>
{
    if (e.HttpClient.Request.RequestUri.AbsolutePath == "/config.pac")
    {
        const string pac = @"
function FindProxyForURL(url, host) {
  if (dnsDomainIs(host, ""mydomain.com""))
    return ""PROXY 127.0.0.1:8000"";
  return ""DIRECT"";
}";
        e.Ok(pac, new[] { new HttpHeader("Content-Type", "application/x-ns-proxy-autoconfig") });
    }
};
```

Then point the OS/browser at `http://127.0.0.1:8000/config.pac` (WinINET / Firefox proxy settings / `ProxyOverride` etc.). Only `mydomain.com` hits Titanium; everything else goes DIRECT.

There's no Fiddler-style `HookUsingPACFile()` helper built-in ? serving + configuring the PAC is the way.

Closing this out.
